docs(api/app-config): add caution on state config (#205)
americanexpress/one-app: docs/api/modules/App-Configuration.md

The `provideStateConfig` directive is useful for supplying string-based key value settings per runtime (e.g. `client` or `server`) and per `environmentLevel` (e.g. QA, Prod, etc). The `environmentLevel` is specified in the [`ONE_CONFIG_ENV` environment variable](../server/Environment-Variables.md#one_config_env) when running the Server.

**⚠️ Please Consider**

> Server specific values used within state config are not isolated from the client side, as they are used
> while rendering server side Holocron modules. Please use caution not to expose any sensitive data in a

> > Holocron module when using `state.get('config')` server-side and rendering markup in your module.

Reviewer: ```suggestion
> Holocron module while using `state.get('config')` for any server side rendering in your module.
```
